kolibri/core/auth/serializers.py

```diff
@@ -63,6 +66,26 @@ def save(self, **kwargs):
 
     def validate(self, attrs):
         username = attrs.get("username")
+        username_validator = RegexValidator(
+            r'[\s`~!@#$%^&*()\-+={}\[\]\|\\\/:;"\'<>,\.\?]',
+            "Enter a valid username. This value can contain only letters, numbers, and underscores.",
+            code=error_constants.INVALID,
+            inverse_match=True,
+        )
+        try:
+            username_validator(username)
+        except DjangoValidationError as e:
+            raise serializers.ValidationError({"username": e.message})
+
+        try:
+            MaxLengthValidator(
```

Comment:
This could also be defined in models.py and reused.

Reply:
Done